scalactic.dotty/src/main/scala/org/scalactic/anyvals/NumericString.scala

```diff
@@ -554,7 +554,7 @@ final class NumericString private (val value: String) extends AnyVal {
    * @return         array of strings produced by splitting `NumericString` at every location matching supplied `regex`, up to `limit` occurrences
    */
   def split(regex: String, limit: Int): Array[String] =
-    value.split(regex, limit)
+    value.split(regex, limit).map(_.nn)
```

Comment:
**Java method returns array**
Java method returns array of type `Array[Null | T]`. We need to map it to `Array[T]`.